File: RustSideStory/SampleSource/error_handling.swift
// No Result enum in Swift standard library
// https://github.com/antitypical/Result

// try/catch but without the performance penalties

do {
	try functionThatThrows()
	print("Function succeeded!")
}
catch let error { // error adopts ErrorType protocol
	print("Found error: \(error)")
}

let response = try? functionThatThrows() ?? "Default Value"
let response = try! functionThatThrows()

// Translates traditional Objective C errors into Swift try/catch

File: RustSideStory/SampleSource/error_handling.rs
// Has the Result enum in the standard library

enum Result<T, E> { // E adopts the Error trait
    Ok(T),
    Err(E),
}

match "5".parse::<i32>() {
	Ok(num) => println!("Found number: {}", num),
	Err(err) => println!("Found error: {}", err),
}

// Or use the try! macro
let result = try!("notanumber".parse::<i32>());
